src/syft/core/common/module.py

```diff
@@ -23,7 +27,9 @@ def __init__(self) -> None:
 
     # this is how we catch the modules being set during subclass init
     def __setattr__(self, name: str, value: Union[Any, "Module"]) -> None:
-        if "torch.nn.modules" in full_name_with_qualname(klass=type(value)):
+        # bug where torch.nn.modules isnt the full name on some imports
```

Comment:
Q: For what case is this happening?

Reply:
Yeah this is weird, and its part of the FQN bug which we need to address separately. I haven't pinned this down to an exact situation but it seems to depend on when you import the library, for example this didn't happen in the tests but it did happen in my jupyter notebook. I think we need to decide exactly what we will support in something like Sy Module so that we can track and filter these the same way that the pytorch Module does. I have added this requirement to the existing ticket: https://github.com/OpenMined/PySyft/issues/4729

Reply:
And the related bug:
https://github.com/OpenMined/PySyft/issues/4745